src/WidgetBase.ts

```diff
@@ -271,17 +285,42 @@ export class WidgetBase<P extends WidgetProperties> extends Evented implements W
 			delete this._previousProperties[propertyName];
 		});
 
-		const diffPropertiesResult = this.diffProperties(this._previousProperties, properties);
-		this._properties = assign(diffPropertiesResult.properties, diffPropertyResults);
+		const diffPropertiesDecorators = this.getDecorator('diffProperties');
+		if (diffPropertiesDecorators.length) {
+			const diffDecorator = diffPropertiesDecorators[diffPropertiesDecorators.length - 1];
```

Comment:
Right, I wanted to take the "last" diffProperties, so it can be overridden. My thinking was if you had extended a widget, that already had a `@diffProperties`, but you added a new property that had special logic, you could just give your own `@diffProperties` as well. Now that I type that out though it doesn't make sense because you could just use a single `@diffProperty` ...